Client/mods/deathmatch/logic/CClientGame.cpp

```diff
@@ -624,12 +624,6 @@ bool CClientGame::StartGame(const char* szNick, const char* szPassword, eServerT
             std::string strUser;
             pBitStream->Write(strUser.c_str(), MAX_SERIAL_LENGTH);
 
-            if (g_pNet->CanServerBitStream(eBitStreamVersion::Discord_InitialImplementation))
```

Comment:
Is it safe, when client and server have different versions?

Reply:
I just checked. We should bump the bitstream in this PR (something like `Discord_Cleanup`). 

If the server has `Discord_InitialImplementation` but not `Discord_Cleanup`, then the client should send an empty string (otherwise player join will fail).

If server has `Discord_Cleanup` then we don't need to send anything.

As for the server -> client bitstream, that doesn't seem to matter. 

For the Lua definition, `setPlayerDiscordJoinParams`, the server requests an RPC - `SET_DISCORD_JOIN_PARAMETERS` - which won't exist if the client is newer than server.

It also tries to send a `CServerInfoSyncPacket` (`PACKET_ID_SERVER_INFO_SYNC`) on player join with flag `SERVER_INFO_FLAG_ALL`, but there isn't a case for handling that flag in `CPacketHandler::Packet_ServerInfoSync`. So if the client is newer, the packet is ignored, and if the client is outdated nothing happens currently anyway.